src/constructors.jl

```diff
@@ -72,96 +72,119 @@ err_bad_iteration_parameter(p) =
-the iteration parameter automatically selected. If
-`retrain=false` (default) and `resampling` is not `nothing`, then
-`iterated_model` behaves like the original model trained on a subset
-of the provided data.
-
-Controlled iteration can be continued with new `fit!` calls (warm
-restart) by mutating a control, or by mutating the iteration parameter
-of `model`, which is otherwise ignored.
-
-
-### Training
-
-Given an instance `iterated_model` of `IteratedModel`, calling
-`fit!(mach)` on a machine `mach = machine(iterated_model, data...)`
-performs the following actions:
-
-- Assuming `resampling !== nothing`, the `data` is split into *train* and
-  *test* sets, according to the specified `resampling` strategy, which
-  must have type `Holdout`.
-
-- A clone of the wrapped model, `iterated_model.model`, is bound to
-  the train data in an internal machine, `train_mach`. If `resampling
-  === nothing`, all data is used instead. This machine is the object
-  to which controls are applied. For example, `Callback(fitted_params
-  |> print)` will print the value of `fitted_params(train_mach)`.
+    IteratedModel(model;
+        controls=MLJIteration.DEFAULT_CONTROLS,
+        resampling=Holdout(),
+        measure=nothing,
+        retrain=false,
+        advanced_options...,
+    )
+
+Wrap the specified supervised `model` in the specified iteration `controls`. Here `model`
+should support iteration, which is true if (`iteration_parameter(model)` is different from
+`nothing`.
+
+Available controls: $CONTROLS_LIST.
+
+!!! important
+
+    To make out-of-sample losses available to the controls, the wrapped `model` is only
+    trained on part of the data, as iteration proceeds. The user may want to force
+    retraining on all data after controlled iteration has finished by specifying
+    `retrain=true`. See also "Training", and the `retrain` option, under "Extended help"
+    below.
+
+# Extended help
+
+# Options
+
+- `controls=$DEFAULT_CONTROLS`: Controls are summarized at
+  [https://JuliaAI.github.io/MLJ.jl/dev/getting_started/](https://JuliaAI.github.io/MLJ.jl/dev/controlling_iterative_models/)
+  but query individual doc-strings for details and advanced options. For creating your own
+  controls, refer to the documentation just cited.
+
+- `resampling=Holdout(fraction_train=0.7)`: The default resampling holds back 30% of data
+  for computing an out-of-sample estimate of performance (the "loss") for controls such
+  as `WithLossDo` and stopping criterion; specify `resampling=nothing` if all data is to
+  be used for controlled iteration, with each out-of-sample loss replaced by the most
+  recent training loss, assuming this is made available by the model
+  (`supports_training_losses(model) == true`). If the model does not provide report a
+  training loss, you can use `resampling=InSample()` instead, with an additional
+  performance cost.  Otherwise, `resampling` must have type `Holdout` or be a vector with
```

Comment:
```suggestion
  training loss, you can use `resampling=InSample()` instead, with an additional
  performance cost.  Otherwise, `resampling` must have type `Holdout` or be a vector with
```
Here by Performance cost. Do you mean there is a slowdown or did you mean that a loss (measure) must be specified? In the latter case I don't think it should be a must since MLJ can infer the default loss to use for a given model in most cases.

Reply:
I removed the confusing ", with an additional performance cost. "